### PR TITLE
Don't allow backend_url to be set in Datatrust.register

### PR DESCRIPTION
**Pull request by rbharath**
Working towards a fix for https://github.com/computablelabs/goest/issues/141. More details once ready to merge.

### CODE REVIEW

**Comment by rbharath** [comment]
This is now ready for review. It removes the setting of `self.backend_url` in `Datatrust.register` and slightly modifies `Datatrust.resolveRegistration` to clear the old backend url if a new datatrust is approved.

This PR fixes one of the old tests and adds a new test to check that multiple registrations work correctly.

File: tests/datatrusttest/register_functionality_test.go [hunk 1]
```diff
@@ -11,19 +11,35 @@ import (
 )
 
 func TestRegister(t *testing.T) {
-	// we may already have a backend registered
-	preUrl, _ := deployed.DatatrustContract.GetBackendUrl(nil)
-	if len(preUrl) == 0 {
+	preAddress, _ := deployed.DatatrustContract.GetBackendAddress(nil)
+	t.Log(fmt.Sprintf("preAddress is: %v", preAddress))
```
Comment (robrobbins) [inline tests/datatrusttest/register_functionality_test.go]: this looks unnecessary

File: tests/datatrusttest/register_functionality_test.go [hunk 2]
```diff
@@ -11,19 +11,35 @@ import (
...
+	backend_address := context.AuthBackend.From
+	t.Log(fmt.Sprintf("backend_address is: %v", backend_address))
```
Comment (robrobbins) [inline tests/datatrusttest/register_functionality_test.go]: same

File: tests/datatrusttest/register_functionality_test.go [hunk 3]
```diff
@@ -11,19 +11,35 @@ import (
...
+	// The backend is already registered
+	if preAddress == backend_address {
+		t.Log(fmt.Sprintf("Addresses match!"))
```
Comment (robrobbins) [inline tests/datatrusttest/register_functionality_test.go]: same

File: tests/datatrusttest/register_functionality_test.go [hunk 4]
```diff
@@ -11,19 +11,35 @@ import (
...
+		// Set datatrust utl
+		_, setErr := deployed.DatatrustContract.SetBackendUrl(test.GetTxOpts(context.AuthBackend, nil,
+			big.NewInt(test.ONE_GWEI*2), 500000), "http://www.icanhazbackend.com")
+		test.IfNotNil(t, setErr, fmt.Sprintf("Error setting backend url: %v", setErr))
+		context.Blockchain.Commit()
+	} else {
+		// The backend is not yet registered
+		preUrl, _ := deployed.DatatrustContract.GetBackendUrl(nil)
+		t.Log(fmt.Sprintf("preUrl is: %v", preUrl))
```
Comment (robrobbins) [inline tests/datatrusttest/register_functionality_test.go]: same

File: tests/datatrusttest/register_functionality_test.go [hunk 5]
```diff
@@ -11,19 +11,35 @@ import (
...
 
-		// url will be stored as we wait on the voting
+		// url will not be stored as we wait on the voting
 		url, _ := deployed.DatatrustContract.GetBackendUrl(nil)
-		if !strings.Contains(url, "icanhazbackend") {
-			t.Errorf("expected url to be %v, got: %v", "http://www.icanhazbackend.com", url)
+		if strings.Contains(url, "icanhazbackend") {
+			t.Errorf("Url was improperly updated to: %v", url)
 		}
+		t.Log(fmt.Sprintf("url was not updated and is still: %v", url))
```
Comment (robrobbins) [inline tests/datatrusttest/register_functionality_test.go]: same

File: tests/datatrusttest/register_functionality_test.go [hunk 6]
```diff
@@ -34,6 +50,34 @@ func TestRegister(t *testing.T) {
 	}
 }
 
+func TestReRegister(t *testing.T) {
+	// Test what happens when a new party tries to register as datatrust
+
+	// The backend is not yet registered
+	preUrl, _ := deployed.DatatrustContract.GetBackendUrl(nil)
+	t.Log(fmt.Sprintf("preUrl is: %v", preUrl))
```
Comment (robrobbins) [inline tests/datatrusttest/register_functionality_test.go]: same

File: tests/datatrusttest/register_functionality_test.go [hunk 7]
```diff
@@ -34,6 +50,34 @@ func TestRegister(t *testing.T) {
...
+		big.NewInt(test.ONE_GWEI*2), 500000), "http://www.thenewbackend.com")
+	test.IfNotNil(t, regErr, fmt.Sprintf("Error registering for backend status: %v", regErr))
+	context.Blockchain.Commit()
+
+	// url will not be stored as we wait on the voting
+	url, _ := deployed.DatatrustContract.GetBackendUrl(nil)
+	if strings.Contains(url, "thenewbackend") {
+		t.Errorf("Url was improperly updated to: %v", url)
+	}
+	t.Log(fmt.Sprintf("url was not updated and is still: %v", url))
```
Comment (robrobbins) [inline tests/datatrusttest/register_functionality_test.go]: same

File: tests/datatrusttest/register_functionality_test.go (hunk 1 above)
Comment (rbharath) [reply tests/datatrusttest/register_functionality_test.go]: Removed.

File: tests/datatrusttest/register_functionality_test.go (hunk 2 above)
Comment (rbharath) [reply tests/datatrusttest/register_functionality_test.go]: Removed

File: tests/datatrusttest/register_functionality_test.go (hunk 3 above)
Comment (rbharath) [reply tests/datatrusttest/register_functionality_test.go]: Removed

**Comment by rbharath** [comment]
Removed all the logging statements. This is ready for review again @robrobbins

**Pull request merged**